File: gasStation/gasStation.py
from typing import List
#you only do var = var = value in python since dec and def happen at same time
class Solution:
    def canCompleteCircuit(self, gas: List[int], cost: List[int]) -> int:
        total = start = tank = 0
        for i in range(len(gas)):
            DIFF = gas[i] - cost[i]
            total += DIFF
            tank += DIFF
            if(tank < 0):
                tank = 0
                start = i+1
        if(total >= 0):
            return start
        return -1

File: gasStation/gasStation.js
/**
 * @param {number[]} gas
 * @param {number[]} cost
 * @return {number}
 */
var canCompleteCircuit = function(gas, cost) {
    let tank = 0, start = 0, total = 0;
    for(let i = 0; i < gas.length; ++i){
        const diff = gas[i] - cost[i];
        total += diff;
        tank += diff;
        if(tank < 0){
            tank = 0;
            start = i+1;
        }
    }
    if(total >= 0)
        return start;
    return -1;
};